app/javascript/components/tree-view/reducers/oncheck.js

```diff
@@ -0,0 +1,95 @@
+import Tree from '@manageiq/react-ui-components/dist/wooden-tree';
+
+const getSelectedKeys = state => Object.values(state).filter(entry => entry.state && entry.state.checked).map(entry => entry.attr.key);
+
+// FIXME: Move the urls outside the component
+const checkUrl = (node, type = '') => {
+  switch (type) {
+    case 'generic':
+      return `${ManageIQ.tree.checkUrl}${encodeURIComponent(node.attr.key)}?check=${+node.state.checked}`;
+    case 'cucollection':
+    case 'sections':
+      return `${ManageIQ.tree.checkUrl}?id=${encodeURIComponent(node.attr.key)}&check=${!!node.state.checked}`;
```

Comment:
The `generic above is basically the same as this one, you can merge it. Having a `/x` or `/?id=x` is the same in Rails

Reply:
Also I'd feel safer if you'd use a real URL builder instead of string concatenation. I'm pretty sure JS has one by default.

Reply:
Also, note the `!!` and the `+` before the `node.state.checked` - one route needs `0 or 1` the other `true or false`

Reply:
That I can fix :wink:

Reply:
Okay, this one actually doesn't care what's being sent in the `check`, so you can do this:
```suggestion
    case 'generic':
    case 'cucollection':
    case 'sections':
      return `${ManageIQ.tree.checkUrl}?id=${encodeURIComponent(node.attr.key)}&check=${+node.state.checked}`;
```